substrate/utils/wasm-builder/src/wasm_project.rs

```diff
@@ -110,78 +110,111 @@ fn crate_metadata(cargo_manifest: &Path) -> Metadata {
 ///
 /// # Returns
 ///
-/// The path to the compact WASM binary and the bloaty WASM binary.
+/// The path to the compact runtime binary and the bloaty runtime binary.
 pub(crate) fn create_and_compile(
 	project_cargo_toml: &Path,
 	default_rustflags: &str,
 	cargo_cmd: CargoCommandVersioned,
 	features_to_enable: Vec<String>,
-	wasm_binary_name: Option<String>,
+	bloaty_blob_name: Option<String>,
 	check_for_runtime_version_section: bool,
 ) -> (Option<WasmBinary>, WasmBinaryBloaty) {
-	let wasm_workspace_root = get_wasm_workspace_root();
-	let wasm_workspace = wasm_workspace_root.join("wbuild");
+	let runtime_workspace_root = get_wasm_workspace_root();
+	let runtime_workspace = runtime_workspace_root.join("wbuild");
 
 	let crate_metadata = crate_metadata(project_cargo_toml);
 
 	let project = create_project(
 		project_cargo_toml,
-		&wasm_workspace,
+		&runtime_workspace,
 		&crate_metadata,
 		crate_metadata.workspace_root.as_ref(),
 		features_to_enable,
 	);
 
-	let profile = build_project(&project, default_rustflags, cargo_cmd);
-	let (wasm_binary, wasm_binary_compressed, bloaty) =
-		compact_wasm_file(&project, profile, project_cargo_toml, wasm_binary_name);
+	let build_config = BuildConfiguration::detect(&project);
+
+	// Build the bloaty runtime blob
+	build_bloaty_blob(&build_config.blob_build_profile, &project, default_rustflags, cargo_cmd);
+
+	// Get the name of the bloaty runtime blob.
+	let default_bloaty_blob_name = get_blob_name(project_cargo_toml);
+	let bloaty_blob_name = bloaty_blob_name.unwrap_or_else(|| default_bloaty_blob_name.clone());
+
+	let bloaty_blob_binary =
+		copy_bloaty_blob(&project, &build_config.blob_build_profile, &bloaty_blob_name);
+
+	// Try to compact and compress the bloaty blob, if the *outer* profile wants it.
+	//
+	// This is because, by default the inner profile will be set to `Release` even when the outer
+	// profile is `Debug`, because the blob built in `Debug` profile is too slow for normal
+	// development activities.
+	let (compact_blob_path, compact_compressed_blob_path) =
+		match build_config.outer_build_profile.wants_compact() {
+			true => {
```

Comment:
`match`ing on a `bool` seems strange, why not an `if`? Will reduce one level of indentation.